pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -103,23 +83,20 @@ int kretprobe__tcp_sendmsg(struct pt_regs *ctx) {
     return handle_message(&t, sent, 0, CONN_DIRECTION_UNKNOWN, packets_out, packets_in, PACKET_COUNT_ABSOLUTE, skp);
 }
 
-SEC("kprobe/tcp_cleanup_rbuf")
-int kprobe__tcp_cleanup_rbuf(struct pt_regs *ctx) {
-    struct sock *sk = (struct sock *)PT_REGS_PARM1(ctx);
-    int copied = (int)PT_REGS_PARM2(ctx);
-    if (copied <= 0) {
-        return 0;
-    }
+SEC("kprobe/tcp_recvmsg/pre_4_1_0")
+int kprobe__tcp_recvmsg__pre_4_1_0(struct pt_regs* ctx) {
     u64 pid_tgid = bpf_get_current_pid_tgid();
-    log_debug("kprobe/tcp_cleanup_rbuf: pid_tgid: %d, copied: %d\n", pid_tgid, copied);
-
-    conn_tuple_t t = {};
-    if (!read_conn_tuple(&t, sk, pid_tgid, CONN_TYPE_TCP)) {
+    log_debug("kprobe/tcp_recvmsg: pid_tgid: %d\n", pid_tgid);
+    int flags = (int)PT_REGS_PARM6(ctx);
+    if (flags & MSG_PEEK) {
         return 0;
     }
 
-    handle_tcp_stats(&t, sk, 0);
-    return handle_message(&t, 0, copied, CONN_DIRECTION_UNKNOWN, 0, 0, PACKET_COUNT_NONE, sk);
+    void *parm2 = (void*)PT_REGS_PARM2(ctx);
+    struct sock* skp;
+    bpf_probe_read_with_telemetry(&skp, sizeof(skp), &parm2);
```

Comment:
wondering why you are using bpf_probe_read_with_telemetry here, while we don't do that in the equivalent `tcp_sendmsg_pre_4_1_0`

Reply:
Removed as this actually did not work on 4.4.